src/core/class.application.php

```diff
@@ -46,61 +57,59 @@ public function __construct(config $config,
      */
     public function start()
     {
-
-        $config = $this->config; // Used in template
-        $settings = $this->settings; //Used in templates to show app version
-        $login = $this->login;
-        $frontController = $this->frontController;
-        $language = $this->language;
+        //Only run telemetry when logged in
         $telemetryResponse = false;
 
+        $this->loadHeaders();
+
         //Override theme settings
         $this->overrideThemeSettings();
 
-        $this->loadHeaders();
+        //Check if Leantime is installed
+        $this->checkIfInstalled();
+
+        //Allow a limited set of actions to be public
+
+
+        if($this->auth->logged_in()===true) {
+
+            //Send telemetry if user is opt in and if it hasn't been sent that day
+            $telemetryResponse = $this->reportService->sendAnonymousTelemetry();
 
-        ob_start();
+            // Check if trying to access twoFA code page, or if trying to access any other action without verifying the code.
+            if($_SESSION['userdata']['twoFAEnabled'] && $_SESSION['userdata']['twoFAVerified'] === false){
 
-        if($this->login->logged_in()===false) {
+                if($this->frontController::getCurrentRoute() !== "twoFA.verify"
+                    && $this->frontController::getCurrentRoute() !== "auth.logout"
+                    && $this->frontController::getCurrentRoute() !== "api.i18n") {
```

Comment:
`if(!in_array($this->frontController::getCurrentRoute(), ['twoFA.verify','auth.logout','api.i18n']))` might be cleaner than 3 separate if checks. Not 100% sure though